src/pool/mempool.rs

```diff
@@ -58,42 +71,77 @@ impl AccountManager {
     {
         let accounts = self.accounts.clone();
 
-        rt_handle.spawn(async move {
+        let handle: JoinHandle<Result<(), eyre::Report>> = rt_handle.spawn(async move {
             loop {
                 for address in &accounts {
-                    Self::process_transaction(address, eth_provider);
+                    let account_filter =
+                        EthDatabaseFilterBuilder::<filter::OzAccount>::default().with_oz_address(address).build();
+
+                    let oz_account = eth_provider
+                        .database()
+                        .get_one::<StoredOzAccount>(account_filter.clone(), None)
+                        .await?
+                        .ok_or_else(|| eyre::eyre!("Account not found in the database"))?;
+
+                    // Convert the optional Ethereum block ID to a Starknet block ID.
+                    let starknet_block_id = eth_provider.to_starknet_block_id(Some(BlockId::default())).await?;
+
+                    // Create a new `ERC20Reader` instance for the Starknet native token
+                    let eth_contract = ERC20Reader::new(*STARKNET_NATIVE_TOKEN, eth_provider.starknet_provider());
+
+                    // Call the `balanceOf` method on the contract for the given address and block ID, awaiting the result
+                    let span = tracing::span!(tracing::Level::INFO, "sn::balance");
+                    let res = eth_contract.balanceOf(address).block_id(starknet_block_id).call().instrument(span).await;
+
+                    if contract_not_found(&res) || class_hash_not_declared(&res) {
+                        return Err(eyre::eyre!("Contract not found or class hash not declared"));
+                    }
+
+                    // Otherwise, extract the balance from the result, converting any errors to ExecutionError
+                    let balance = res.map_err(ExecutionError::from)?.balance;
+
+                    // Convert the low and high parts of the balance to U256
+                    let low: U256 = into_via_wrapper!(balance.low);
+                    let high: U256 = into_via_wrapper!(balance.high);
+
+                    // Combine the low and high parts to form the final balance and return it
+                    let balance = low + (high << 128);
+
+                    if oz_account.current_tx_hash.is_none() && balance > U256::from(u128::pow(10, 18)) {
+                        Self::process_transaction(address, eth_provider);
+                    }
                 }
 
                 tokio::time::sleep(Duration::from_secs(1)).await;
             }
         });
+
+        rt_handle.spawn(async {
+            if let Err(e) = handle.await {
+                tracing::error!("Error in spawned task: {:?}", e);
+            }
+        });
+
+        Ok(())
     }
 
-    fn process_transaction<SP>(address: &Felt, eth_provider: &EthDataProvider<SP>)
+    fn process_transaction<SP>(_address: &Felt, eth_provider: &EthDataProvider<SP>)
```

Comment:
Yes, it is the dedicated function that will execute the transaction by taking an OpenZeppelin account. For now I left it empty because we don't have accounts etc and so we couldn't really test but in a follow up I planned to add the execution of the transaction in it.